src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Threading/Dispatcher.cs

```diff
@@ -2170,46 +2170,39 @@ private bool GetMessage(ref MSG msg, IntPtr hwnd, int minMessage, int maxMessage
             // Win32 GetMessage().
             bool result;
             UnsafeNativeMethods.ITfMessagePump messagePump = GetMessagePump();
```

Comment:
Seems like there is manual counting going on here. I think this needs to be disposed of still. I don't see `GetMessagePump` in this review though so could be mistaken.

Reply:
`TfThreadMgrWrapper` implements `ITfMessagePump` and I obtain all these pointers inside wrapper. Once wrapper would be removed, message pump would be removed too. I think that's valid since QI - Release pair used exactly the same count of times.

Reply:
I see the pattern now. This appears okay. Thanks.